src/js/utils/cancelable.ts

```diff
@@ -1,9 +1,14 @@
-export default function cancelable(callback) {
+type Cancelable = {
+    async: (...args: [any]) => Promise<any>;
+    cancel: () => void;
+    cancelled: () => boolean;
+}
+
+export default function cancelable(callback: (result?: any) => any): Cancelable {
```

Comment:
Do cancelable callbacks always return something? Might need to be => any | void

Reply:
I thought that would be covered by `any` since it disables type checks.

Reply:
I know @zetagame and I had briefly chatted about `any` and `void` and after looking a bit more into things, it seems I was wrong in assuming `void` was not of type `any`.

[Void Type](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#325-the-void-type)

> The Void type is a subtype of the Any type and a supertype of the Null and Undefined types, but otherwise Void is unrelated to all other types.

So this would be fine.